### PR TITLE
fix(tests): recompute nemotron-nas rotary buffers in HF phase of checkpoint robustness

### PR DESCRIPTION
**Pull request by adil-a**
## Summary
- Phase 4 of `test_checkpoint_robustness_llm.py` reloads the trained model via plain `transformers.AutoModelForCausalLM` and compares logits against the training reference. For `model_type="nemotron-nas"` (and `"gemma3"`), rotary `inv_freq` is a non-persistent buffer computed in `__init__` and not written to the checkpoint — transformers 5.x default meta-device init never runs the computation on a real device, so the buffer contains uninitialized GPU memory. Attention rotates Q/K by garbage frequencies, diverging the HF reload from the training reference.
- `nemo-automodel`'s own loader already handles this via `_reinit_non_persistent_buffers` (allow-listed for `"nemotron-nas"` and `"gemma3"`) inside `apply_model_infrastructure`. The robustness test's HF path did not call it, so Phase 4 was measuring a broken HF model.
- This patch calls the same reinit helper after every HF `from_pretrained` site in Phase 4 (PEFT + SFT, both `hf_device_map_auto` branches) via `_reinit_rotary_per_module`, which resolves each module's own device so it works correctly under `device_map="auto"`.
- Also carries over the `timeout_minutes: 20` bump and `resume_loss_threshold: 5e-2` from the now-closed #1951; the `hf_kl_threshold` bump in that PR is no longer needed and stays at the standard `5e-3`.

## Verification
Full robustness run on `nvidia/Llama-3_3-Nemotron-Super-49B-v1_5` (command matches `scripts/finetune_launcher.sh`'s `ROBUSTNESS_CMD` template):

```
[Memory]   Peak VRAM 16.93 GB, CPU RSS 2.92 GB
[Phase 3]  Automodel-from-consolidated max KL = 0.00e+00  (threshold 0.00)           PASS
[Fused QKV] No combined projection keys in adapter (874 keys checked)                PASS
[Phase 4]  HF-loaded max KL = 9.17e-04                    (threshold 5.00e-03)       PASS
[Phase 6]  Step 5 diff=1.71e-02, Step 6 diff=3.04e-02, Step 7 diff=1.91e-02
           Training resumption verified (3 steps compared)                           PASS
================= 1 passed, 28 warnings in 1459.42s (0:24:19) ==================
```

Prior to this fix the same Phase 4 produced max KL ~**1.05e+01** against the same reference — roughly **11,000x worse**, which is why #1951 had been raising `hf_kl_threshold` from `5e-3` to `1.5e1` to mask the loader bug.

## Test plan
- [x] Full `finetune_launcher.sh` robustness test on `llama3_3_nemotron_super_49B_squad_peft.yaml` passes with standard `hf_kl_threshold: 5e-3`.
- [x] Per-module device handling verified on a 2-layer DeciLM load forced onto multiple GPUs via `device_map="auto"` with `max_memory` caps — rotary buffers land on each module's own device after reinit, forward pass produces finite logits without cross-device errors.
- [ ] Smoke-test other `nemotron-nas` / `gemma3` recipes whose robustness thresholds had been inflated (none observed on main).

Closes #1951

### CODE REVIEW

**Comment by adil-a** [comment]
Superseded by #1984. Your test-harness change (recompute nemotron-nas rotary buffers via `_reinit_non_persistent_buffers` in Phase 4) and the `llama3_3_nemotron_super_49B_squad_peft.yaml` changes (`timeout_minutes: 20`, `resume_loss_threshold: 5e-2`) are cherry-picked onto #1984. Verified still passes on cw-dfw with Flash SFT + PEFT after merge (both bit-exact / under threshold). Thanks for root-causing the nemotron-nas rotary init issue.
